lib/kubernixos.nix

```diff
@@ -19,26 +20,30 @@ let
       };
     };
 
+  items = lib.mapAttrs merge cfg.manifests;
+
 in
-{
-
-  kubernixos = {
-
-    config = (removeAttrs cfg ["assertions" "manifests"]) // { checksum = kubernixos; };
-
-    manifests = with pkgs;
-    let
-      # Assertion validation borrowed from /modules/system/activation/top-level.nix
-      failedAssertions = with pkgs.lib; map (x: x.message) (filter (x: !x.assertion) cfg.assertions);
-    in
-      if failedAssertions != []
-      then throw "\nFailed assertions:\n${concatStringsSep "\n" (map (x: "- ${x}") failedAssertions)}"
-      else
-      {
-        apiVersion = "v1";
-        kind = "List";
-        items = lib.mapAttrsToList merge cfg.manifests;
+rec{
+  eval = {
+      config = (removeAttrs cfg ["manifests" "schemas"]) // { checksum = kubernixos; };
+
+      manifests = {
+          apiVersion = "v1";
+          kind = "List";
+          items = lib.attrValues items;
       };
-
   };
+
+  build =
+      pkgs.runCommand "kubernixos-${kubernixos}" { nativeBuiltInputs = [pkgs.kubeval cfg.schemas]; } ''
+          mkdir -p $out
+          ${lib.concatStringsSep "\n" (lib.mapAttrsToList (n: i: "echo '${builtins.toJSON i}' >$out/${n}.json") items)}
```

Comment:
What happens here if a json string contains a single quote? Accept build failure and move on?

Reply:
Alternatively one could build `writeText` each json snippet and cat them together, or create both the individuals and the sum in Nix before writing it out. No reason to go through bash strings.

Reply:
Have a look at: a245a1d4db17ca810cbc4c595eb5f8c2d26b04df

Reply:
:ok_hand: